tests/unity/unity_test.cmake

```diff
@@ -0,0 +1,135 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
+
+set(UNITY_PRODUCTS_DIR "${APPLICATION_BINARY_DIR}/unity")
+set(CMOCK_PRODUCTS_DIR "${UNITY_PRODUCTS_DIR}/mocks")
+set(CMOCK_DIR "${ZEPHYR_BASE}/../cmock")
+set(RUBY_EXEC "/usr/bin/ruby")
+file(MAKE_DIRECTORY "${UNITY_PRODUCTS_DIR}")
+
+# Generate cmock for provided header file.
+function(cmock_generate header_path dst_path)
+  file(MAKE_DIRECTORY "${dst_path}")
+
+  execute_process(
+    COMMAND ${RUBY_EXEC}
+    ${CMOCK_DIR}/lib/cmock.rb
+    --mock_prefix=mock_
+    --plugins=return_thru_ptr\;ignore_arg\;ignore\;callback\;array
+    --mock_path=${dst_path}
+    ${header_path}
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    RESULT_VARIABLE op_result
```

Comment:
Should we abort when op_result != 0 ?

Reply:
added fatal_error on errors returned by commands.